OrbitLinuxTracing/PerfEventRingBuffer.cpp

```diff
@@ -85,32 +85,36 @@ bool PerfEventRingBuffer::HasNewData() {
 }
 
 void PerfEventRingBuffer::ReadHeader(perf_event_header* header) {
-  assert(IsOpen());
   ReadAtTail(reinterpret_cast<uint8_t*>(header), sizeof(perf_event_header));
   assert(header->type != 0);
   assert(metadata_page_->data_tail + header->size <= metadata_page_->data_head);
 }
 
 void PerfEventRingBuffer::SkipRecord(const perf_event_header& header) {
-  assert(IsOpen());
   // Write back how far we read from the buffer.
   metadata_page_->data_tail += header.size;
 }
 
 void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header,
                                         void* record) {
-  assert(IsOpen());
   ReadAtTail(static_cast<uint8_t*>(record), header.size);
   SkipRecord(header);
 }
 
+void PerfEventRingBuffer::ReadRawAtOffset(void* dest, uint64_t offset,
```

Comment:
Should this be a uint8_t* that is passed in?

Reply:
I don't believe so, the reason for void* is that you can pass any pointer without a cast, which is  what you want here.

Reply:
Actually making it clear that you are reading a count number of _bytes_ might be a good idea, but should it be uint8_t* or char*?

Reply:
There are two versions of that method now, one takes uint8_t* and one char*.